substratevm/src/com.oracle.objectfile/src/com/oracle/objectfile/debugentry/MethodEntry.java

```diff
@@ -72,26 +74,51 @@ public String getParamName(int idx) {
         return paramNames[idx];
     }
 
-    public boolean match(String methodName, String paramSignature, String returnTypeName) {
-        if (!methodName.equals(this.memberName)) {
-            return false;
+    public int compareTo(String methodName, String paramSignature, String returnTypeName) {
+        if (!memberName.equals(methodName)) {
+            return memberName.compareTo(methodName);
         }
-        if (!returnTypeName.equals(valueType.getTypeName())) {
-            return false;
+        if (!valueType.getTypeName().equals(returnTypeName)) {
+            return valueType.getTypeName().compareTo(returnTypeName);
         }
         int paramCount = getParamCount();
         if (paramCount == 0) {
-            return paramSignature.trim().length() == 0;
+            return 0 - paramSignature.trim().length();
         }
         String[] paramTypeNames = paramSignature.split((","));
         if (paramCount != paramTypeNames.length) {
-            return false;
+            return paramCount - paramTypeNames.length;
         }
         for (int i = 0; i < paramCount; i++) {
             if (!paramTypeNames[i].trim().equals(getParamTypeName(i))) {
-                return false;
+                return getParamTypeName(i).compareTo(paramTypeNames[i].trim());
             }
         }
-        return true;
+        return 0;
+    }
+
+    public boolean isDeoptTarget() {
+        return isDeoptTarget;
+    }
+
+    @Override
+    public int compareTo(MethodEntry other) {
+        assert other != null;
+        int result = methodName().compareTo(other.methodName());
```

Comment:
This could also profit from the strategy shown in https://github.com/oracle/graal/pull/3304/files#r627333745

Reply:
I personally prefer this over https://github.com/oracle/graal/pull/3304/files#r627333745